caas/kubernetes/provider/k8s.go

```diff
@@ -199,12 +212,17 @@ func newK8sBroker(
 		apiextensionsClientUnlocked: apiextensionsClient,
 		dynamicClientUnlocked:       dynamicClient,
 		envCfgUnlocked:              newCfg.Config,
-		namespace:                   newCfg.Name(),
-		modelUUID:                   modelUUID,
-		newWatcher:                  newWatcher,
-		newStringsWatcher:           newStringsWatcher,
-		newClient:                   newClient,
-		randomPrefix:                randomPrefix,
+		informerFactory: informers.NewSharedInformerFactoryWithOptions(
```

Comment:
we need manage `informerFactory` like `clientUnlocked, apiextensionsClientUnlocked , etc`.
Because we may refresh those clients here https://github.com/juju/juju/blob/develop/caas/kubernetes/provider/k8s.go#L311.

Reply:
Will take a look

Reply:
Done & Done